sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -457,7 +457,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    */
   @scala.annotation.varargs
   def text(paths: String*): Dataset[String] = {
-    format("text").load(paths : _*).as[String](sparkSession.implicits.newStringEncoder)
+    format("text").load(paths : _*).select("value")
```

Comment:
Yeah, I think the semantic of dropping the columns is okay (I'm not sure the other option makes sense to me).  I'm a little concerned that we are still doing partition discovery, but I guess removing that is only an optimization.